### DISCUSSION TITLE
With Nginx getting CORS issue when I click to add/edit or delete a tracking domain.

### DISCUSSION THREAD

**Discussion by ghost** [postalserver/postal#2893] [Help with using Postal] [answered]
If I access the server at http://mail03.mydomain.com:5000 everything works, but if I do https://mail03.mydomain.com, everything seems to work, until I go to add/edit/delete a tracking domain. When it posts, the network response is "Unprocessable Entity". I'm not sure what's wrong.

Looking at the log, this appears to be a CORS style issue:
"_message=HTTP Origin header (https://mail03.mydomain.com) didn't match request.base_url (http://mail03.mydomain.com)"

My nginx config is:
```
server {
        listen 443 ssl default_server;
        listen [::]:443 ssl default_server;
        server_name mail03.mydomain.com;
        ssl_certificate /etc/ssl/certs/server.pem;
        ssl_certificate_key /etc/ssl/private/server.key;

        location / {
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header Host $host;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_redirect off;
                proxy_pass http://192.168.1.125:5224;
        }
}

server {
        listen 443 ssl;
        listen [::]:443 ssl;
        server_name click03.mydomain.com;
        ssl_certificate /etc/ssl/certs/server.pem;
        ssl_certificate_key /etc/ssl/private/server.key;

        location / {
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header Host $host;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_redirect off;
                proxy_pass http://192.168.1.125:5224;
        }
}
```

My postal.yml has:
version: 2

```
web_server:
  default_bind_address: 192.168.1.125

```

I can post more, but I've already had to mask the domain, I can post more if I need to.

Thank you in advance!

**Comment by ghost** [comment, marked as answer]
To answer my own question, once I knew it was a CORS error I tried adding:
`X-Forwarded-Proto https; `
to the nginx configuration and it worked!

**Reply by willpower232** [reply]
yeah the four headers from #2254 tend to go hand in hand, when you need one, you probably need all of them :sweat_smile:
